macros/timeHistoryAnalyzer.py

```diff
@@ -0,0 +1,158 @@
+                      help="Maximum fraction of masked channel, over the whole detector, for a scan to be considered")
+
+    # Configuration of the range finding algo
+    parser.add_option("--numEndScans", type=int, dest="numEndScans", default=5,
+                      help="Number of 'good' scans to end a range")
+    parser.add_option("--minBadScans", type=int, dest="minBadScans", default=4,
+                      help="Minimum number of 'bad' scans to keep a range")
+    # Only for 'zeroInputCap' range finder
+    parser.add_option("--minNoise", type=float, dest="minNoise", default=0.0414,
+                      help="Lower bound on noise for the 'zeroInputCap' range finder")
+    parser.add_option("--maxNoise", type=float, dest="maxNoise", default=0.109,
+                      help="Upper bound on noise for the 'zeroInputCap' range finder")
+
+    (options, args) = parser.parse_args()
+
+    if options.inputDir is None:
+        print("Error: The -i argument is required")
+        sys.exit(os.EX_USAGE)
+
+    if not os.path.isdir(options.inputDir):
+        print("Error: Not a directory: %s" % options.inputDir)
+        sys.exit(os.EX_USAGE)
+
+    from gempython.gemplotting.utils.anahistory import (
+        findRangesMaskReason,
+        findRangesMask,
+        findRangesZeroInputCap,
+        TimeSeriesData)
+
+    findRangesFct = None
+    findRangesKwArgs = {
+        'numEndScans': options.numEndScans,
+        'minBadScans': options.minBadScans,
+    }
+    if options.ranges == "mask":
+        findRangesFct = findRangesMask
+    elif options.ranges == "maskReason":
+        findRangesFct = findRangesMaskReason
+    elif options.ranges == "zeroInputCap":
+        findRangesFct = findRangesZeroInputCap
+        findRangesKwArgs['minNoise'] = options.minNoise
+        findRangesKwArgs['maxNoise'] = options.maxNoise
+    else:
+        print("Error: Invalid argument for --ranges: %s " % options.ranges)
+        sys.exit(os.EX_USAGE)
+
+    data = TimeSeriesData(options.inputDir)
+    data.removeBadScans(minAverageNoise = options.minScanAvgNoise,
+                        maxMaskedStripOrChanFraction = options.maxScanMaskedFrac)
+
+    from gempython.gemplotting.utils.anaInfo import MaskReason
+
+    # Find ranges
+    ranges = [] # [vfat][stripOrChan][ranges]
+    for vfat in range(24):
+        ranges.append([])
+        for stripOrChan in range(128):
+            ranges[vfat].append(findRangesFct(data, vfat, stripOrChan, **findRangesKwArgs))
+            pass
+        pass
+
+    # Filter if needed
+    if options.onlyCurrent:
+        for vfat in range(24):
+            for stripOrChan in range(128):
+                ranges[vfat][stripOrChan] = list(filter(lambda r: r.end == data.numScans(),
+                                                        ranges[vfat][stripOrChan]))
+
+    # Initialize tables
+    rangesTables = [ [] for vfat in range(24) ]
```

Comment:
A function exists for this, see [getEmptyPerVFATList](https://github.com/cms-gem-daq-project/gem-plotting-tools/blob/develop/anautilities.py#L89).  It is not necessary to change but you may find it useful.